packages/backend-ai/src/conversation/runtime/conversation_turn_response_sender.ts

```diff
@@ -0,0 +1,75 @@
+import { ConversationTurnEvent } from './types.js';
+import { ContentBlock } from '@aws-sdk/client-bedrock-runtime';
+
+type MutationResponseInput = {
+  input: {
+    conversationId: string;
+    content: ContentBlock[];
+    associatedUserMessageId: string;
+  };
+};
+
+/**
+ * TODO docs
+ */
+export class ConversationTurnResponseSender {
+  /**
+   * TODO docs
+   */
+  constructor(
+    private readonly event: ConversationTurnEvent,
+    private readonly _fetch = fetch
+  ) {}
+
+  sendResponse = async (message: ContentBlock[]) => {
+    const request = this.createMutationRequest(message);
+    const res = await this._fetch(request);
```

Comment:
Shouldn't there be a try/catch as well since fetch can throw?

Reply:
I refrained from aggressively catching exceptions, following this philosophy.  https://softwareengineering.stackexchange.com/questions/365427/try-catch-log-rethrow-is-anti-pattern .

From practical point of view (that I observed when working on this)
1. When `fetch` throws:
   1. Exception bubbles up all the way to lambda runtime
   2. Exception is logged into CW as is.
   3. Lambda metrics show failed executions.
2. Statements below were added to catch "silent failures":
   1. When request comes back with non-2xx status
   2. When GQL error is in payload with 2xx status (this is horrible btw...)
   3. In both cases fetch was happy and there was no trace of failure.

Reply:
I agree with just re-throwing is bad, but if a better error message can be included I think it's worth it. Since this is a data plane call and going through AppSync, what will customers see when this fails?

Reply:
> if a better error message can be included I think it's worth it

We can do that. But right place for that is going to be in top level component. I.e. add try-catch to this top level messaging https://github.com/aws-amplify/amplify-backend/blob/ae3fbbafbc8e454acddafd990f3057b750ca30fc/packages/backend-ai/src/conversation/runtime/conversation_turn_executor.ts#L27-L30 . I.e explicitly print that conversation handler failed. (following this https://stackoverflow.com/questions/13042975/exception-handling-practices-top-level-vs-every-function )

> Since this is a data plane call and going through AppSync, what will customers see when this fails?

This is a questionable part. As lambda invocation is asynchronous from appsync pov (fire and forget). I.e. frontend will see bot going silent.
This is a tradeoff made in upstream (appsync part) design and there doesn't seem to be 100% good solution visible right now.